torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -319,11 +324,16 @@ class TensorPipeAgent : public RpcAgent {
   // Map to store the expiration times for each message.
   std::map<
       steady_clock_time_point,
-      std::vector<std::pair<
+      std::vector<std::tuple<
+          uint64_t, // messageId
           std::shared_ptr<AtomicJitFuture>,
-          std::chrono::milliseconds>>>
+          std::chrono::milliseconds>>> // timeout
```

Comment:
Nit: for readability it may make sense to define a custom struct instead of this tuple, so that fields can be accessed with meaningful names, rather than with `std::get`.